DigitalOffice/Services/UserService/Broker/Consumers/GetUserInfoConsumer.cs

```diff
@@ -0,0 +1,59 @@
+﻿using System.Threading.Tasks;
+using Broker.Requests;
+using Broker.Responses;
+using LT.DigitalOffice.Kernel.Broker;
+using LT.DigitalOffice.UserService.Database.Entities;
+using LT.DigitalOffice.UserService.Mappers.Interfaces;
+using LT.DigitalOffice.UserService.Repositories.Interfaces;
+using MassTransit;
+using Microsoft.AspNetCore.Mvc;
+
+namespace LT.DigitalOffice.UserService.Broker.Consumers
+{
+    /// <summary>
+    /// Consumer for getting information about the user
+    /// </summary>
+    public class GetUserInfoConsumer : IConsumer<IInternalGetUserInfoRequest>
+    {
+        private readonly IRequestClient<IInternalGetUserPositionRequest> client;
+        private readonly IUserRepository repository;
+        private readonly ITwoModelsMapper<DbUser,IInternalUserPositionResponse, object> mapper;
+
+        public GetUserInfoConsumer(
+            [FromServices] IUserRepository repository,
+            [FromServices] IRequestClient<IInternalGetUserPositionRequest> client,
+            [FromServices] ITwoModelsMapper<DbUser,IInternalUserPositionResponse, object> mapper)
+        {
+            this.repository = repository;
+            this.client = client;
+            this.mapper = mapper;
+        }
+
+        public async Task Consume(ConsumeContext<IInternalGetUserInfoRequest> context)
+        {
+            var response= await client.GetResponse<IOperationResult<IInternalUserPositionResponse>>(
+                new
+                {
+                    context.Message.UserId
+                });
+            if (!response.Message.IsSuccess)
+            {
+                await context.RespondAsync(new
+                {
+                    IsSuccess = false,
+                    Errors = response.Message.Errors
+                });
+            }
```

Comment:
This looks like a double check. I actually don't mind this, so it's up to Artem

Reply:
I forgot to wrap the code below in "else". Why transfer this logic to sender ?

Reply:
Because it is usually done on the sender's side. And why request user id, if this is do it on sender's side, by id to the user, they request user data from the user service, and the check response is done on the sender's side. You have very veiled broker logic.